components/style/properties.mako.rs

```diff
@@ -5838,6 +5838,15 @@ pub enum PropertyDeclaration {
     Custom(::custom_properties::Name, DeclaredValue<::custom_properties::SpecifiedValue>),
 }
 
+impl PropertyDeclaration {
+    #[inline]
+    pub fn id(&self) -> u64 {
+        unsafe {
+            intrinsics::discriminant_value(self)
```

Comment:
This uses the same ID for all custom properties. Maybe return `Atom` instead? (With static atoms for non-custom properties)